Displays user information for sender of message on u hotkey

israelsgalaxy opened the pull request:
<!-- See README for documentation, or ask in #zulip-terminal if unclear -->
### What does this PR do, and why?
This PR adds functionality which displays user information for the sender of a message when the 'u' hotkey is pressed on an active message box. This would match the behavior in the web app.

### External discussion & connections
<!-- [x] all that apply, specifying topic and adding numbers after # for issues/PRs -->
- [ ] Discussed in **#zulip-terminal** in `topic`
- [x] Fully fixes #1397
- [ ] Partially fixes issue #
- [ ] Builds upon previous unmerged work in PR #
- [ ] Is a follow-up to work in PR #
- [ ] Requires merge of PR #
- [ ] Merge will enable work on #

### How did you test this?
<!-- [x] all that apply -->
- [x] Manually - Behavioral changes
- [x] Manually - Visual changes
- [x] Adapting existing automated tests
- [ ] Adding automated tests for new behavior (or missing tests)
- [ ] Existing automated tests should already cover this (*only a refactor of tested code*)

### Self-review checklist for each commit
- [x] It is a [minimal coherent idea](https://github.com/zulip/zulip-terminal#structuring-commits---speeding-up-reviews-merging--development)
- [x] It has a commit summary following the [documented style](https://github.com/zulip/zulip-terminal#structuring-commits---speeding-up-reviews-merging--development) (title & body)
- [x] It has a commit summary describing the motivation and reasoning for the change
- [x] It individually passes linting and tests
- [ ] It contains test additions for any new behavior
- [ ] It flows clearly from a previous branch commit, and/or prepares for the next commit

### Visual changes    <!-- DELETE SECTION IF NO VISUAL CHANGE -->
![Updated UI](https://github.com/zulip/zulip-terminal/assets/59661679/79df3003-397a-4c43-bd1e-fbe44926219f)

<!-- NOTE: Attached videos/images will be clearer from smaller terminal windows -->

neiljp commented on zulipterminal/core.py:
```diff
@@ -312,9 +312,9 @@ def show_about(self) -> None:
         )
 
-    def show_user_info(self, user_id: int) -> None:
+    def show_user_info(self, user_id: int, cmd: str = "USER_INFO") -> None:
```
This doesn't need to be a default value, and would be clearer if it were not.

There's also no need to abbreviate the text to 'cmd' - presumably this is command?

This is a refactor commit - it doesn't change behavior - so it's good to note it as such as an additional commit title area.

neiljp reviewed (commented):
@israelsgalaxy Thanks for looking at another issues - this is functional :+1:

Exploring using this, my only thought is that it may be useful to change the title of the popup to represent that this is the information of the message sender, not simply that of a selected user as before.

One of the things which may be contributing to my feeling about this is that now we have a popup coming from two different places (user list, message list) the key to exit this popup is no longer unique. So from a user perspective, it's not clear which key to use, and it isn't shown in the popup even though it differs. This suggests working on #1002 would be useful.

israelsgalaxy commented:
Thank you for the review @neiljp . I have changed the title of the popup to give users more clarity. I have also removed the default value in the show_user_info method.

neiljp commented:
@israelsgalaxy Thanks for the adjustment - I reordered the commits and slightly adjusted the commit text for accuracy and have pushed back here, and will merge now CI is complete :tada:

Other than #1002, as mentioned above, a more directly connected follow-up that would relate to this would be to explore turning the user information details in the message information into a single button which would also open on `u` from there, much like links. A simpler first approach would be to show it (and allow it to work) in the 'Viewing Actions' list there. It would likely be useful to turn those into buttons that work as part of the overall list of scrollable items which can be activated. So the less simple path of turning the user information into a button would fit with that direction, particularly since it already has a place in the popup.
